src/librustc_trans/back/symbol_names.rs

```diff
@@ -226,7 +239,11 @@ pub fn internal_name_from_type_and_suffix<'a, 'tcx>(ccx: &CrateContext<'a, 'tcx>
                                                     -> String {
     let path = [token::intern(&t.to_string()).as_str(),
                 gensym_name(suffix).as_str()];
-    let hash = get_symbol_hash(ccx, &Vec::new(), cstore::LOCAL_CRATE, &[t]);
+    let def_path = DefPath {
+        data: vec![],
+        krate: cstore::LOCAL_CRATE,
+    };
```

Comment:
In the long term we should move to `<T as Trait>::method`, at least for trait-related shims.